server/routerlicious/packages/routerlicious-base/src/test/nexus/io.spec.ts

```diff
@@ -463,6 +467,115 @@ describe("Routerlicious", () => {
 						]);
 						await Promise.all(promises);
 					});
+
+					it("Verify v2 signals", async () => {
+						let userSignalCounts = [0, 0, 0];
+
+						const createSignalPromise = (
+							userIndex: number,
+							socket: LocalWebSocket,
+						): Promise<ISignalMessage> => {
+							return new Promise((resolve) => {
+								socket.on("signal", (signal: ISignalMessage) => {
+									userSignalCounts[userIndex]++;
+									resolve(signal);
+								});
+							});
+						};
+						const firstSocket = webSocketServer.createConnection();
+
+						const firstConnectMessage = await connectToServer(
+							testId,
+							testTenantId,
+							testSecret,
+							firstSocket,
+							true,
+						);
+
+						const secondSocket = webSocketServer.createConnection();
+
+						const secondConnectMessage = await connectToServer(
+							testId,
+							testTenantId,
+							testSecret,
+							secondSocket,
+							true,
+						);
+
+						const thirdSocket = webSocketServer.createConnection();
+
+						await connectToServer(testId, testTenantId, testSecret, thirdSocket, true);
+
+						const user1SignalPromise = createSignalPromise(0, firstSocket);
+						const user2SignalPromise = createSignalPromise(1, secondSocket);
+						const user3SignalPromise = createSignalPromise(2, thirdSocket);
+
+						const targetedSignal: ISentSignalMessage = {
+							targetClientId: firstConnectMessage.clientId,
+							content: "TestSignal",
+						};
```

Comment:
I am not sure what client side API ends up looking and what user injects here, but if we end up having an additional field here beyond "content", what happens? Does server do any type checking? Would it be worthwhile testing it? Generally speaking, just want to encourage you to test malformed, unexpected, and edge cases. Edge case can be a behavior case.

Reply:
The server does type checking for ISentSignalMessage, I added a test case for malformed signals that include invalid properties.